.travis.yml

```diff
@@ -212,14 +222,74 @@ jobs:
             - g++-5
             - libc6-dev-i386
       before_install:
-        - mkdir bin ; ln -s /usr/bin/gcc-5 bin/gcc
+        - mkdir bin
+        - ln -s /usr/bin/gcc-5 bin/gcc
+        - ln -s /usr/bin/g++-5 bin/g++
       install:
         - ccache -z
         - ccache --max-size=1G
         - cmake -H. -Bbuild '-DCMAKE_BUILD_TYPE=Release'  '-DCMAKE_CXX_COMPILER=/usr/bin/g++-5'
         - cmake --build build -- -j4
       script: (cd build; ctest -V -L CORE -j2)
 
+    # cmake build using g++-7
+    - stage: Test different OS/CXX/Flags
+      os: linux
+      compiler: gcc
+      cache: ccache
+      env:
+        - BUILD_SYSTEM=cmake
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          packages:
+            - g++-7
+            - libc6-dev-i386
+      before_install:
+        - mkdir bin
+        - ln -s /usr/bin/gcc-7 bin/gcc
+        - ln -s /usr/bin/g++-7 bin/g++
+      install:
+        - ccache -z
+        - ccache --max-size=1G
+        - cmake -H. -Bbuild '-DCMAKE_BUILD_TYPE=Release'  '-DCMAKE_CXX_COMPILER=/usr/bin/g++-7'
+        - cmake --build build -- -j4
+      script: (cd build; ctest -V -L CORE -j2)
+
+    # cmake build using clang++-6
+    - stage: Test different OS/CXX/Flags
+      os: linux
+      compiler: clang
+      cache: ccache
+      env:
+        - BUILD_SYSTEM=cmake
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-trusty-6.0
+          packages:
+            - libwww-perl
+            - g++-5
+            - clang-6.0
+            - libstdc++-5-dev
+            - libubsan0
+            - parallel
+            - libc6-dev-i386
+      before_install:
+        - mkdir bin
+        # Use gcc/g++ 5 for tests, as Clang doesn't work yet
```

Comment:
What is it that's going wrong - could you be more verbose about this, please?

Reply:
Could you please include a reference to #2370 in the comment? (Though we should of course really fix #2370...)

Reply:
Already done, rotate your eyeballs down one line ;)

Reply:
Somehow GitHub should get smarter about this. (Surely it can't be me who is lacking the smarts :-P)